libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_UDP_IP.c

```diff
@@ -280,7 +280,8 @@ UDPPacket_t *pxUDPPacket = (UDPPacket_t *) pxNetworkBuffer->pucEthernetBuffer;
 				destinationAddress.sin_port = usPort;
 				destinationAddress.sin_addr = pxUDPPacket->xIPHeader.ulDestinationIPAddress;
 
-				if( xHandler( ( Socket_t ) pxSocket, ( void* ) pcData, ( size_t ) pxNetworkBuffer->xDataLength,
+				/* The value of 'xDataLength' was proven to be at least the size of a UDP packet in prvProcessIPPacket(). */
+				if( xHandler( ( Socket_t ) pxSocket, ( void* ) pcData, ( size_t ) ( pxNetworkBuffer->xDataLength - ipUDP_PAYLOAD_OFFSET_IPv4 ),
```

Comment:
FYI we recently found an integer overflow case in the unit test for `ulDNSHandlePacket()` and [fixed it by explicitly checking](https://github.com/aws/amazon-freertos/pull/1615) that `pxNetworkBuffer->xDataLength >= sizeof( UDPPacket_t )`.  Will we see another failing unit test if we don't do that check prior to line 284 here?

Reply:
Sorry, I hadn't see your earlier PR. Thanks for pointing out. I will respond directly to it.
Thanks, Hein

Reply:
I'm confused by the implication of the change in behavior from the perspective of the callee. Is this within the specification of xHandler?

Reply:
@htibosch As Gary pointed out do we need to make sure that ```pxNetworkBuffer->xDataLength >= ipUDP_PAYLOAD_OFFSET_IPv4```  before doing ``` pxNetworkBuffer->xDataLength - ipUDP_PAYLOAD_OFFSET_IPv4 ``` i.e.

```c
if( pxNetworkBuffer->xDataLength >= ipUDP_PAYLOAD_OFFSET_IPv4 )
{
    /* Call xHandler. */
}
else
{
    /* Handle malformed packet. */
}
```

Reply:
@aggarg, @htibosch 's comment on PR #1615 points out that this was validated in `prvProcessIPPacket()`, so we are probably okay here.  That PR was in response to a failing unit test in which `ulDNSHandlePacket()` was called directly, so that condition hadn't been guaranteed.  We'll move forward with this and monitor the unit test results for any regressions.